Thanks — Prettier's happy now, but CI is red on a **unit test**: `src/serializer/serializer.test.ts > Serializer > Handles various function definitions` fails with `AssertionError: expected Promise{…} to deeply equal Promise{…}` (1 failed / 290 passed).

That "Promise deeply equal Promise" shape usually means a `deserialize(...)` result isn't being awaited, so the assertion compares two unresolved Promises — likely a spot where the new `options` argument was threaded through but the call lost its `await` (or the inline-snapshot value needs regenerating). Could you run `yarn vitest run src/serializer/serializer.test.ts` locally, fix that case, and push? Everything else (design, opt-in gating, the new reject test) looks great — this is the last thing between us and merging the RCE fix.

--- a/typescript/src/serializer/serializer.test.ts
+++ b/typescript/src/serializer/serializer.test.ts
@@ -155,14 +155,18 @@ describe("Serializer", () => {
       verifyDeserialization(fn, target);
 
       if (R.isFunction(fn) && R.isFunction(target)) {
-        const l = fn("A");
-        const r = target("A");
-
+        // Await both sides before comparing: some of these functions are
+        // async, and comparing unresolved Promise objects with toEqual is
+        // structurally unreliable (Promises have no own enumerable
+        // properties, so the assertion's outcome can depend on timing
+        // rather than the actual resolved value).
+        const l = await fn("A");
+        const r = await target("A");
         if (R.isFunction(l) && R.isFunction(r)) {
           // @ts-expect-error intended
-          expect(l("B")).toEqual(r("B"));
+          expect(await l("B")).toEqual(await r("B"));
         } else {
-          expect(fn("A")).toEqual(target("A"));
+          expect(l).toEqual(r);
         }
       } else {
         expect(fn).toBe(target);

